### PR TITLE
tools: show me the pgmap in the mon db with ceph_monstore_tool

### PR DESCRIPTION
**Pull request by songbaisen**
tools: show me the pgmap in the mon db with ceph_monstore_tool

Signed-off-by:song baisen song.baisen@zte.com.cn

### CODE REVIEW

**Comment by xiexingguo** [comment]
The failure is true. 
Can you check?
```
[ 83%] Building CXX object src/rgw/CMakeFiles/rgw_a.dir/rgw_acl_swift.cc.o
CMakeFiles/ceph-monstore-tool.dir/ceph_monstore_tool.cc.o: In function `read_pgmap_full(PGMap&, MonitorDBStore*)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/ceph_monstore_tool.cc:527: undefined reference to `PGMap::update_pg(pg_t, ceph::buffer::list&)'
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/ceph_monstore_tool.cc:535: undefined reference to `PGMap::update_osd(int, ceph::buffer::list&)'
CMakeFiles/ceph-monstore-tool.dir/ceph_monstore_tool.cc.o: In function `main':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/ceph_monstore_tool.cc:766: undefined reference to `PGMap::dump(std::ostream&) const'
collect2: error: ld returned 1 exit status
make[3]: *** [bin/ceph-monstore-tool] Error 1
make[2]: *** [src/tools/CMakeFiles/ceph-monstore-tool.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....

**Comment by songbaisen** [comment]
@xiexingguo  Oh thank you! I will fix this later.

**Comment by songbaisen** [comment]
@tchaikov  Hello sir. Here look like have a build bug  .It can build success in my own environmental like below. But it can not build success in the community environmental like below.

My own environmental build result:
build succeeded.
make[1]: Leaving directory `/home/gitceph/ceph/man'
Making all in doc
make[1]: Entering directory `/home/gitceph/ceph/doc'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/gitceph/ceph/doc'
Making all in systemd
make[1]: Entering directory `/home/gitceph/ceph/systemd'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/gitceph/ceph/systemd'
Making all in selinux
make[1]: Entering directory `/home/gitceph/ceph/selinux'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/gitceph/ceph/selinux

The community environmental build result:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/ceph_monstore_tool.cc:527: undefined reference to `PGMap::update_pg(pg_t, ceph::buffer::list&)'

File: src/tools/Makefile-server.am [hunk 1]
```diff
@@ -3,7 +3,7 @@ ceph_osdomap_tool_LDADD = $(LIBOS) $(CEPH_GLOBAL) $(BOOST_PROGRAM_OPTIONS_LIBS)
 bin_DEBUGPROGRAMS += ceph-osdomap-tool
 
 ceph_monstore_tool_SOURCES = tools/ceph_monstore_tool.cc
-ceph_monstore_tool_LDADD = $(LIBOS) $(CEPH_GLOBAL) $(BOOST_PROGRAM_OPTIONS_LIBS)
+ceph_monstore_tool_LDADD = $(LIBOS) $(CEPH_GLOBAL) $(BOOST_PROGRAM_OPTIONS_LIBS) $(LIBMON_TYPES)
```
Comment (tchaikov) [inline src/tools/Makefile-server.am]: you need to update the CMakeLists.txt accordingly.

Comment (songbaisen) [inline src/tools/Makefile-server.am]: @tchaikov  Thank you sir.Learn it.

**Comment by tchaikov** [comment]
monitor will not keep track of pgmap after moving this responsibility to ceph-mgr. so closing now.

**Pull request closed without merging**